### PR TITLE
drivers/nrf24l01p: fixes

### PR DESCRIPTION
**Pull request by PeterKietzmann**
This PR 
1. fixes one small error in the nrf24l01p initialization code (flush tx **and** rx fifo)
2. changes the API of `nrf24l01p_set_power` a bit. This corrects the (then) faulty function call (by value). With this change the function looks more consistent to the other functions and the NULL pointer check inside becomes superfluous. 
3. extends the test application so return values are checked for errors

### CODE REVIEW

**Comment by LudwigOrtmann** [comment]
Is it reasonable to split the API/type change from the fix?

**Comment by PeterKietzmann** [comment]
I assume not. To be honest I just did'nt think about this so much. I reverted this API change

File: drivers/nrf24l01p/nrf24l01p.c [hunk 1]
```diff
@@ -653,10 +653,6 @@ int nrf24l01p_set_power(nrf24l01p_t *dev, int *pwr)
 
     nrf24l01p_read_reg(dev, REG_RF_SETUP, &rf_setup);
 
-    if (pwr == NULL) {
-        return -1;
-    }
-
```
Comment (LudwigOrtmann) [inline drivers/nrf24l01p/nrf24l01p.c]: Why is this check unnecessary now?

**Comment by LudwigOrtmann** [comment]
"fix type error" is not really fitting the change, is it?

**Comment by PeterKietzmann** [comment]
Yes, maybe you're right. Any ideas for naming the commit?

**Comment by BytesGalore** [comment]
how about fix initial setup?

**Comment by PeterKietzmann** [comment]
I changed the commit messagt as proposed by @BytesGalore. Think it's fine

**Comment by LudwigOrtmann** [comment]
What about https://github.com/RIOT-OS/RIOT/pull/2330#discussion_r23228350 ?

**Comment by PeterKietzmann** [comment]
@LudwigOrtmann, here were some things that got confused (by my side), sorry for that. Please read the updated PR description and have a look at the two commits.

**Comment by LudwigOrtmann** [comment]
Well, except for the overlong commit message in https://github.com/PeterKietzmann/RIOT/commit/4ec73bf9b2248217c2a646b7828f5e7c850cb314 it's looking good regarding the form ;)

However, I just tried to run the test application on an `stm32f4discovery` and all the registers are `11111111` or `ff ff ff ff ff`. Before I start rechecking all the connections - did you test this?

**Comment by PeterKietzmann** [comment]
I shortened the overlong commit message a bit. Also I adapted the README as it was outdated a bit. Of course I tested this and it worked fine. I get the same results as stated in the [README](https://github.com/RIOT-OS/RIOT/tree/master/tests/driver_nrf24l01p_lowlevel). Oftend I got results as you described when there were connection errors

**Comment by LudwigOrtmann** [comment]
OK, I just thought maybe the recent SPI changes might have led to new surprising behavior... will recheck connections.

**Comment by PeterKietzmann** [comment]
Nope, thats not the error. This is the point where I stared after #2317 was merged...

**Comment by LudwigOrtmann** [comment]
At least one of the pins was off, checking the rest ;)

**Comment by LudwigOrtmann** [comment]
Hm, no luck. Are you using the default configuration from the `Makefile` or using your own pin assignments?

**Comment by PeterKietzmann** [comment]
I'm using the default from the `Makefile`:
```
SPI_PORT ?= SPI_0
CE_PIN   ?= GPIO_0
CS_PIN   ?= GPIO_1
IRQ_PIN  ?= GPIO_2
```

**Comment by LudwigOrtmann** [comment]
Oh shoot! I've been using the wrong SPI port... using the right port everything works =)

**Comment by LudwigOrtmann** [comment]
Sorry for the hold-up, ACK.

**Comment by PeterKietzmann** [comment]
And go

**Pull request merged**